base/sparse/cholmod.jl

```diff
@@ -1242,11 +1242,7 @@ function cholfact!{Tv}(F::Factor{Tv}, A::Sparse{Tv}; shift::Real=0.0)
 end
 
 """
-    cholfact!(F::Factor, A::Union{SparseMatrixCSC{<:Real},
-        SparseMatrixCSC{Complex{<:Real}},
-        Symmetric{<:Real,SparseMatrixCSC{<:Real,SuiteSparse_long}},
-        Hermitian{Complex{<:Real},SparseMatrixCSC{Complex{<:Real},SuiteSparse_long}}};
-        shift = 0.0) -> CHOLMOD.Factor
+    cholfact!(F::Factor, A::Union{SparseMatrixCSC{<:Real},SparseMatrixCSC{Complex{<:Real}},Symmetric{<:Real,SparseMatrixCSC{<:Real,SuiteSparse_long}},Hermitian{Complex{<:Real},SparseMatrixCSC{Complex{<:Real},SuiteSparse_long}}}; shift = 0.0) -> CHOLMOD.Factor
```

Comment:
Anyway, this signature is impossible to read already in the docs. Maybe remove the long list of types, possibly using a type alias, and describe the type in the text? The point of docstrings is to be easily readable. For an exact reference, people can call `methods`.

I also wonder why the method for `Sparse` below isn't documented at al.